sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -311,6 +311,19 @@ private[spark] object SQLConf {
     defaultValue = Some(false),
     doc = "When true, enable filter pushdown for ORC files.")
 
+  val HIVE_PARALLEL_FILE_LISTING_ENABLED = booleanConf("spark.sql.hive.parallelFileListing",
+    defaultValue = Some(false),
```

Comment:
Have you test this feature with HDFS, or will it introduce regression for other workload?

If not, we should turn it on by default, or even don't have this config (it's annoying to see more and more configs coming up).